src/webui/webapplication.cpp

```diff
@@ -700,11 +721,40 @@ bool WebApplication::validateHostHeader(const QStringList &domains) const
     }
 
     LogMsg(tr("WebUI: Invalid Host header. Request source IP: '%1'. Received Host header: '%2'")
-           .arg(m_env.clientAddress.toString(), m_request.headers[Http::HEADER_HOST])
+           .arg(m_clientAddress.toString(), m_request.headers[Http::HEADER_HOST])
             , Log::WARNING);
     return false;
 }
 
+QHostAddress WebApplication::resolveClientAddress() const
+{
+    QHostAddress clientAddress = m_env.clientAddress;
+
+    if (!m_isReverseProxySupportEnabled)
+        return clientAddress;
+
+    // Only reverse proxy can overwrite client address
+    if (m_trustedReverseProxyList.contains(clientAddress))
+    {
+        const QString forwardedFor = m_request.headers.value(Http::HEADER_X_FORWARDED_FOR);
+
+        if (!forwardedFor.isEmpty())
+        {
+            // client address is the 1st global IP in X-Forwarded-For or, if none available, the 1st IP in the list
+            const QStringList remoteIpList = forwardedFor.split(",");
+            for (const QString &remoteIp : remoteIpList)
+            {
+                if (clientAddress.setAddress(remoteIp) && clientAddress.isGlobal())
+                    return clientAddress;
+            }
+
+            clientAddress.setAddress(remoteIpList.at(0));
```

Comment:
This is very bad, you are going out of bounds if the list is empty.
Also you should use `[]`.